### PR TITLE
kubeadm: add kinder test jobs for k8s 1.19

### PR DESCRIPTION
**Pull request by RA489**
/hold

for the kubeadm PR to merge first:
https://github.com/kubernetes/kubeadm/pull/2113

/sig cluster-lifecycle

### CODE REVIEW

**Comment by RA489** [comment]
@neolit123 looking for lgtm on this PR.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml [hunk 1]
```diff
@@ -41,6 +41,46 @@ periodics:
...
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-kubeadm
+    testgrid-tab-name: kubeadm-kinder-discovery-1-19
+    testgrid-alert-email: kubernetes-sig-cluster-lifecycle+testgrid@googlegroups.com
+    description: "OWNER: sig-cluster-lifecycle (kinder); Uses kubeadm/kinder to create a cluster and test alternative discovery methods for kubeadm join"
+    testgrid-num-columns-recent: "20"
+    testgrid-num-failures-to-alert: "4"
+    testgrid-alert-stale-results-hours: "48"
+  decoration_config:
+    timeout: 40m
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml:59]: we changed the timeout for all existing jobs to 60m

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml [hunk 2]
```diff
@@ -33,7 +33,47 @@ periodics:
...
+- name: ci-kubernetes-e2e-kubeadm-kinder-upgrade-1-18-1-19
+  interval: 12h
+  decorate: true
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-kubeadm,sig-release-1.19-informing
+    testgrid-tab-name: kubeadm-kinder-upgrade-1-18-1-19
+    testgrid-alert-email: kubernetes-sig-cluster-lifecycle+testgrid@googlegroups.com
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml:53]: please add this email `, kubernetes-release-team@googlegroups.com`

[1]

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-x-on-y.yaml [hunk 3]
```diff
@@ -18,6 +18,46 @@ periodics:
...
+- name: ci-kubernetes-e2e-kubeadm-kinder-1-19-on-1-18
+  interval: 12h
+  decorate: true
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-kubeadm,sig-release-1.19-informing
+    testgrid-tab-name: kubeadm-kinder-1-19-on-1-18
+    testgrid-alert-email: kubernetes-sig-cluster-lifecycle+testgrid@googlegroups.com
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-x-on-y.yaml:53]: [1]

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder.yaml [hunk 4]
```diff
@@ -41,6 +41,46 @@ periodics:
...
+- name: ci-kubernetes-e2e-kubeadm-kinder-1-19
+  interval: 12h
+  decorate: true
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-kubeadm,sig-release-1.19-informing
+    testgrid-tab-name: kubeadm-kinder-1-19
+    testgrid-alert-email: kubernetes-sig-cluster-lifecycle+testgrid@googlegroups.com
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder.yaml:53]: [1]

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml (hunk 1 above)
Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml]: updated to 60m

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml (hunk 2 above)
Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml]: updated.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-x-on-y.yaml (hunk 3 above)
Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-x-on-y.yaml]: updated.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder.yaml (hunk 4 above)
Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder.yaml]: updated

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml [hunk 5]
```diff
@@ -41,6 +41,46 @@ periodics:
...
+- name: ci-kubernetes-e2e-kubeadm-kinder-discovery-1-19
+  interval: 12h
+  decorate: true
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-kubeadm
+    testgrid-tab-name: kubeadm-kinder-discovery-1-19
+    testgrid-alert-email: kubernetes-sig-cluster-lifecycle+testgrid@googlegroups.com, kubernetes-release-team@googlegroups.com
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml:53]: no need for the release-team email here.
it's only needed for the x-on-y, upgrade and regular jobs.

[1]

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-external-etcd.yaml [hunk 6]
```diff
@@ -41,6 +41,46 @@ periodics:
...
+- name: ci-kubernetes-e2e-kubeadm-kinder-external-etcd-1-19
+  interval: 12h
+  decorate: true
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-kubeadm
+    testgrid-tab-name: kubeadm-kinder-external-etcd-1-19
+    testgrid-alert-email: kubernetes-sig-cluster-lifecycle+testgrid@googlegroups.com, kubernetes-release-team@googlegroups.com
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-external-etcd.yaml:53]: [1]

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kustomize.yaml [hunk 7]
```diff
@@ -41,6 +41,46 @@ periodics:
...
+- name: ci-kubernetes-e2e-kubeadm-kinder-kustomize-1-19
+  interval: 12h
+  decorate: true
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-kubeadm
+    testgrid-tab-name: kubeadm-kinder-kustomize-1-19
+    testgrid-alert-email: kubernetes-sig-cluster-lifecycle+testgrid@googlegroups.com, kubernetes-release-team@googlegroups.com
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kustomize.yaml:53]: [1]

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml (hunk 5 above)
Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml]: removed.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-external-etcd.yaml (hunk 6 above)
Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-external-etcd.yaml]: removed.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kustomize.yaml (hunk 7 above)
Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-kustomize.yaml]: removed.

**Comment by RA489** [comment]
@neolit123 updated the PR. PTAL.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml [hunk 8]
```diff
@@ -33,7 +33,47 @@ periodics:
...
+    repo: kubernetes
+    base_ref: release-1.19
+    path_alias: k8s.io/kubernetes
+  - org: kubernetes
+    repo: kubeadm
+    base_ref: master
+    path_alias: k8s.io/kubeadm
+  spec:
+    containers:
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20200420-e830a3a-1.19
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml:71]: looks like the latest kubekins image SHA changed.
all additions in the PR should use `v20200710-6b3b6fe-...` now.

Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade.yaml]: updated the changed SHA.

**Pull request merged**